### PR TITLE
feat: versioning data models

### PR DESCRIPTION
**Pull request by Libertonius**
#41

### CODE REVIEW

File: services/client/src/components/VersionsModal.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+import useTableState from 'hooks/useTableState';
+
+import TableList from './TableList';
+
+import formatDistanceToNow from '../utils/formatDistanceToNow';
+
+const { TextArea } = Input;
+
+const VersionssModal = ({ versions, onRestore }) => {
```
Comment (ilyozzz) [inline services/client/src/components/VersionsModal.js:15]: typo

File: services/client/src/hooks/useSchemas.js [hunk 2]
```diff
@@ -168,22 +172,22 @@ export default (props = {}) => {
...
-      updateMutation,
-      execUpdateMutation,
       exportMutation,
       execExportMutation,
-      batchMutation,
-      execBatchMutation,
+      branchMutation,
+      execBranchMutation,
+      versionMutation,
+      execVersionMutation,
```
Comment (ilyozzz) [inline services/client/src/hooks/useSchemas.js:188]: such naming is a bit confused, suggestion:
```
- createBranchMutation
- execCreateBranchMutation
- createVersionMutation
- execCreateVersionMutation
```

File: services/hasura/metadata/tables.yaml [hunk 3]
```diff
@@ -60,6 +60,9 @@
           user:
             id:
               _eq: X-Hasura-User-Id
+- table:
+    name: migrations
+    schema: auth
```
Comment (ilyozzz) [inline services/hasura/metadata/tables.yaml:65]: pls don't track this table. it breaks metadata in some cases

File: services/hasura/migrations/1677672007088_create_table_public_branches/up.sql [hunk 4]
```diff
@@ -0,0 +1,18 @@
+CREATE TABLE "public"."branches" ("id" uuid NOT NULL DEFAULT gen_random_uuid(), "created_at" timestamptz NOT NULL DEFAULT now(), "updated_at" timestamptz NOT NULL DEFAULT now(), "name" text NOT NULL DEFAULT 'main', "user_id" uuid NOT NULL, "datasource_id" uuid NOT NULL, PRIMARY KEY ("id") , FOREIGN KEY ("user_id") REFERENCES "public"."users"("id") ON UPDATE cascade ON DELETE cascade, FOREIGN KEY ("datasource_id") REFERENCES "public"."datasources"("id") ON UPDATE cascade ON DELETE cascade, UNIQUE ("name", "datasource_id"));
```
Comment (ilyozzz) [inline services/hasura/migrations/1677672007088_create_table_public_branches/up.sql:1]: maybe you need to delete previous migrations which creates and drops `branches` table via `hasura migrate status` and `hasura migrate delete`

File: services/client/src/hooks/useCurrentUser.js [hunk 5]
```diff
@@ -78,10 +86,11 @@ const currentUserQuery = `
       ${reportFragment}
     }
 
-    dataschemas (
-      order_by: { created_at: desc }
+    branches (
+      order_by: { created_at: desc },
+      where: {status: { _neq: "arhived" }}
```
Comment (ilyozzz) [inline services/client/src/hooks/useCurrentUser.js:91]: archived

File: services/client/src/hooks/useCurrentUser.js [hunk 6]
```diff
@@ -126,13 +135,18 @@ const currentUserWithTeamQuery = `
...
       order_by: { created_at: desc },
-      where: { datasource: {
-        team_id: { _eq: $teamId }
-      } }
+      where: { 
+        datasource: {
+          team_id: { _eq: $teamId }
+        },
+        status: {
+          _neq: "arhived"
```
Comment (ilyozzz) [inline services/client/src/hooks/useCurrentUser.js:145]: archived

File: services/client/src/hooks/useSchemas.js [hunk 7]
```diff
@@ -7,60 +7,36 @@ import useQuery from './useQuery';
...
-    update_dataschemas_by_pk(pk_columns: $pk_columns, _set: $_set) {
-      id
-    }
-  }
-`;
-
 const delSchemaMutation = `
   mutation ($id: uuid!) {
-    delete_dataschemas_by_pk(id: $id) {
+    update_branches_by_pk(_set: {status: "arhived"}, pk_columns: {id: $id}) {
```
Comment (ilyozzz) [inline services/client/src/hooks/useSchemas.js:39]: archived

Comment (ifokeev) [reply services/client/src/hooks/useSchemas.js:39]: also value should be as enum table

**Pull request merged**